app/services/search/vacancy_search.rb

```diff
@@ -36,21 +36,21 @@ def location_search
   end
 
   def wider_search_suggestions
-    return unless vacancies.empty? && search_criteria[:location].present?
+    return unless total_count.zero? && search_criteria[:location].present?
 
-    Search::WiderSuggestionsBuilder.new(search_criteria).suggestions
+    @wider_search_suggestions ||= Search::WiderSuggestionsBuilder.new(search_criteria).suggestions
```

Comment:
Costly operation. (Does between 1 and 9 search queries to DB, each with a different Radius).